components/DetailPictureView/index.js

```diff
@@ -0,0 +1,48 @@
+import { useRouter } from "next/router";
+import BackButton from "../BackButton";
+import Image from "next/image";
+import { styled } from "styled-components";
+
+export default function DetailPictureView({ countries, isCurrentCountry }) {
+  const router = useRouter();
+  const currentPublicId = router.query.detailPictureView;
+
+  if (!currentPublicId) {
+    return null;
+  }
+
+  const { publicIds, imagesUrls } = isCurrentCountry;
+  const publicId = publicIds.findIndex((id) => id === currentPublicId);
+
+  const currentPicture = imagesUrls[publicId];
+
+  if (currentPicture === undefined) {
+    return null;
+  }
```

Comment:
This bit shouldn't be part of any component's logic. Better move it up to the route level, in this case `pages/Subpages/[detailPictureView].js`.

Components should be kept as dumb as possible.